docs/platform-external/use-case-oci-upi.md

```diff
@@ -0,0 +1,1829 @@
+```
+
+!!! tip "Helper - OCI CLI documentation"
+    - [`oci os object put`](https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.29.1/oci_cli_docs/cmdref/os/object/put.html)
+
+    OCI Console path: `Menu > Storage > Buckets > (Choose the Compartment `openshift`) > (Choose the Bucket `openshift-infra`) > Objects > Upload`
+
+- Import to the Instance Image service:
+```sh
+STORAGE_NAMESPACE=$(oci os ns get | jq -r .data)
+oci compute image import from-object -bn $BUCKET_NAME --name images/${IMAGE_NAME} \
+    --compartment-id $COMPARTMENT_ID_OPENSHIFT -ns $STORAGE_NAMESPACE \
+    --display-name ${IMAGE_NAME} --launch-mode "PARAVIRTUALIZED" \
+    --source-image-type "QCOW2"
+
+# Gather the Custom Compute image for RHCOS
+IMAGE_ID=$(oci compute image list --compartment-id $COMPARTMENT_ID_OPENSHIFT \
+  --display-name $IMAGE_NAME | jq -r '.data[0].id')
+```
+!!! tip "Helper"
+    OCI CLI documentation for [`oci compute image import`](https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.29.1/oci_cli_docs/cmdref/compute/image/import/from-object.html)
+
+    OCI CLI documentation for [`oci os ns get`](https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.29.1/oci_cli_docs/cmdref/os/ns/get.html)
+
+#### Bootstrap
+
+The Bootstrap node is responsible to create the temporary control plane, and serve
+the ignition files to other nodes through Machine Config Server.
+
+The OCI user data has a size limitation that prevents to use of the bootstrap
+ignition file directly when launching the node. A new ignition file will be
+created replacing it with a remote URL fetching from the temporary Bucket Object URL.
+
+Once the bootstrap instance is created, it must be attached to the Load Balancer in the
+Backends for Kubernetes API Server and Machine Config Server.
+
+Steps:
+
+- Upload the `bootstrap.ign` to the infrastructure bucket
+
+```sh
+oci os object put -bn $BUCKET_NAME --name bootstrap-${CLUSTER_NAME}.ign \
+    --file $INSTALL_DIR/bootstrap.ign
+```
+
+!!! tip "Helper"
+    OCI CLI documentation for [`oci os object put`](https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.29.1/oci_cli_docs/cmdref/os/object/put.html)
+
+
+- Generate the signed URL for the bootstrap object to be used in the user data:
+
+```sh
+EXPIRES_TIME=$(date -d '+1 hour' --rfc-3339=seconds)
+IGN_BOOTSTRAP_URL=$(oci os preauth-request create --name bootstrap-${CLUSTER_NAME} \
+    -bn $BUCKET_NAME -on bootstrap-${CLUSTER_NAME}.ign \
+    --access-type ObjectRead  --time-expires "$EXPIRES_TIME" \
+    | jq -r '.data["full-path"]')
+```
+
+!!! tip "Helper"
+    OCI CLI documentation for [`oci os preauth-request create`](https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.29.0/oci_cli_docs/cmdref/os/preauth-request/create.html)
+
+The ignition URL for the Bootstrap node must be available in the `$IGN_BOOTSTRAP_URL`.
+
+!!! warning "Attention"
+    The bucket Object URL will expire in one hour if you are planning to create
```

Comment:
i'm not sure what the proper spelling should be, but earlier we capitalize "Bucket Object URL" and here we don't. if this is a proper noun/phrase from oci then we should probably capitalize it everywhere.